phpBB/phpbb/content_visibility.php

```diff
@@ -144,7 +144,15 @@ public function get_count($mode, $data, $forum_id)
 	*/
 	public function is_visible($mode, $forum_id, $data)
 	{
-		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $data[$mode . '_visibility'] == ITEM_APPROVED;
+		$visibility = $data[$mode . '_visibility'];
+		$poster_key = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+		$is_visible = $this->auth->acl_get('m_approve', $forum_id) ||
+			($visibility == ITEM_APPROVED) ||
```

Comment:
This could be moved first, before the `acl_get()` call for performance reasons as it's the most common case.

Reply:
Done